alfred-telemetry-platform/src/main/java/eu/xenit/alfred/telemetry/webscripts/PrometheusWebScript.java

```diff
@@ -33,8 +40,20 @@ public void execute(WebScriptRequest webScriptRequest, WebScriptResponse webScri
             throw new WebScriptException(404, "The global MeterRegistry doesn't contain a PrometheusMeterRegistry");
         }
 
-        webScriptResponse.setStatus(200);
-        writeTextToResponse(PrometheusRegistryUtil.extractPrometheusScrapeData(meterRegistry), webScriptResponse);
+        if (!semaphore.tryAcquire()) {
+            throw new WebScriptException(429, "Max number of active requests (" + maxRequests + ") reached");
```

Comment:
Is 429 appropriate? This indicates that this is a sort of "rate-limit" and a client error. While actually something is wrong server-side. I think a 503 is more appropriate.

Reply:
I agree!